File: qqbot-app/src/main/java/com/mieai/qqbot/app/media/MediaRuntimeConfiguration.java
package com.mieai.qqbot.app.media;

import com.mieai.qqbot.client.FileMediaAssetStore;
import com.mieai.qqbot.client.MediaAssetStore;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration(proxyBeanMethods = false)
@EnableConfigurationProperties(MediaRuntimeProperties.class)
public class MediaRuntimeConfiguration {
    @Bean
    MediaAssetStore mediaAssetStore(MediaRuntimeProperties properties) {
        return new FileMediaAssetStore(properties.getStagingDirectory());
    }
}

File: qqbot-module-qqbot-runtime/src/main/kotlin/com/mieai/qqbot/app/media/MediaRuntimeConfiguration.kt
package com.mieai.qqbot.app.media

import com.mieai.qqbot.client.FileMediaAssetStore
import com.mieai.qqbot.client.MediaAssetStore
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration(proxyBeanMethods = false)
@EnableConfigurationProperties(MediaRuntimeProperties::class)
class MediaRuntimeConfiguration {
    @Bean
    fun mediaAssetStore(properties: MediaRuntimeProperties): MediaAssetStore =
        FileMediaAssetStore(properties.stagingDirectory)
}
